src/compiler/binder.ts

```diff
@@ -1086,52 +1076,52 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
         }
         switch (node.kind) {
             case SyntaxKind.WhileStatement:
-                bindWhileStatement(node as WhileStatement);
+                bindWhileStatement(node);
                 break;
             case SyntaxKind.DoStatement:
-                bindDoStatement(node as DoStatement);
+                bindDoStatement(node);
                 break;
             case SyntaxKind.ForStatement:
-                bindForStatement(node as ForStatement);
+                bindForStatement(node);
                 break;
             case SyntaxKind.ForInStatement:
             case SyntaxKind.ForOfStatement:
                 bindForInOrForOfStatement(node as ForInOrOfStatement);
                 break;
             case SyntaxKind.IfStatement:
-                bindIfStatement(node as IfStatement);
+                bindIfStatement(node);
                 break;
             case SyntaxKind.ReturnStatement:
             case SyntaxKind.ThrowStatement:
-                bindReturnOrThrow(node as ReturnStatement | ThrowStatement);
+                bindReturnOrThrow(node);
                 break;
             case SyntaxKind.BreakStatement:
             case SyntaxKind.ContinueStatement:
                 bindBreakOrContinueStatement(node as BreakOrContinueStatement);

```

Comment:
Shouldn't this cast be unnecessary?

Reply:
https://github.com/microsoft/TypeScript/pull/54148#discussion_r1186601123

(here and the other comments like this one)